executing/_position_node_finder.py

```diff
@@ -541,27 +545,28 @@ def node_match(node_type, **kwargs):
 
         raise VerifierFailure(title, node, instruction)
 
-    def instruction(self, index):
+    def instruction(self, index: int) -> dis.Instruction:
         return self.bc_list[index // 2]
 
-    def opname(self, index):
+    def opname(self, index: int) -> str:
         return self.instruction(index).opname
 
     def find_node(
         self,
-        index,
-        match_positions=("lineno", "end_lineno", "col_offset", "end_col_offset"),
-        typ=(ast.expr, ast.stmt, ast.excepthandler, ast.pattern),
-    ):
+        index: int,
+        match_positions: Sequence[str]=("lineno", "end_lineno", "col_offset", "end_col_offset"),
+        typ: tuple[Type, ...]=(ast.expr, ast.stmt, ast.excepthandler, ast.pattern),
+    ) -> EnhancedAST:
         position = self.instruction(index).positions
+        assert position is not None and position.lineno is not None
 
         return only(
-            node
+            cast(EnhancedAST, node)
             for node in self.source._nodes_by_line[position.lineno]
             if isinstance(node, typ)
             if not isinstance(node, ast.Expr)
             # matchvalue.value has the same positions as matchvalue themself, so we exclude ast.MatchValue
-            if not isinstance(node, ast.MatchValue)
+            if not isinstance(node, ast.Match)
```

Comment:
?

Reply:
Hmm. I think this was needed in earlier Python versions, and when I was testing this file with <3.11, but given it only works with 3.11+ and the other versions now exclude it from the mypy checking, I'll revert it.